src/CBLDocument.cc

```diff
@@ -246,20 +246,25 @@ static UnretainedValueToBlobMap& newBlobs() {
 
 
 void CBLDocument::registerNewBlob(CBLNewBlob* blob) {
-    newBlobs().useLocked()->insert({blob->properties(), blob});
+    newBlobs().useLocked()->insert({blob->digest(), blob});
 }
 
 
 void CBLDocument::unregisterNewBlob(CBLNewBlob* blob) {
-    newBlobs().useLocked().get().erase(blob->properties());
+    newBlobs().useLocked().get().erase(blob->digest());
 }
 
 
 CBLNewBlob* CBLDocument::findNewBlob(FLDict dict) {
     return newBlobs().useLocked<CBLNewBlob*>([dict](auto &newBlobs) -> CBLNewBlob* {
-        auto i = newBlobs.find(dict);
-        if (i == newBlobs.end())
+        auto digest = Dict(dict)[kCBLBlobDigestProperty].asString();
+        assert(digest);
+        auto i = newBlobs.find(digest);
+        if (i == newBlobs.end()) {
+            CBL_Log(kCBLLogDomainDatabase, kCBLLogWarning,
```

Comment:
This will also log a warning if the same blob is encountered multiple times.  If that is a rare situation then this is probably ok but otherwise we should attempt to track only real times that this is an issue.

Reply:
That is true. It's a rare situation but it is a misleading warning message. So I think we just don't log it here. We already log when blob is installed and I think that is enough for tracking.

Reply:
OK I added the warning back with additional info saying that "blob might have already been installed" to cover the OK case.